app/models.js

```diff
@@ -1,85 +1,116 @@
 const db = require("../db/connection");
 
 exports.selectAllTopics = () => {
-    return db.query(`SELECT * FROM topics;`)
-    .then(({rows}) => {
-        return rows
-    })
+  return db.query(`SELECT * FROM topics;`).then(({ rows }) => {
+    return rows;
+  });
 };
 
 exports.selectArticleById = (id) => {
-    return db.query(`SELECT * FROM articles WHERE article_id = $1;`, [id])
-    .then(({rows}) => {
-        if (rows.length === 0) {
-            return Promise.reject({status: 404, msg: "Article does not exist"})
-        } else {
-            return rows[0]
-        }
-    })
-}
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1;`, [id])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Article does not exist" });
+      } else {
+        return rows[0];
+      }
+    });
+};
 
-exports.selectAllArticles = () => {
-    return db.query(`SELECT articles.author, title, articles.article_id, articles.topic, articles.created_at, articles.votes, article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC;`)
-    .then(({rows}) => {
-        return rows
-    })
+exports.selectAllArticles = (topic) => {
+  let queryStr = `SELECT articles.author, title, articles.article_id, articles.topic, articles.created_at, articles.votes, article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id `;
+  const queryParams = []
+  const validTopics = ["mitch", "cats", "paper"];
+  if (validTopics.includes(topic)) {
+    queryStr += `WHERE articles.topic = $1 `;
+    queryParams.push(topic)
+  }
+  queryStr += `GROUP BY articles.article_id ORDER BY created_at DESC;`;
+  return db.query(queryStr, queryParams).then(({ rows }) => {
```

Comment:
The development database will have more topics than this. So we will need to do an SQL query on the topics table to make sure that it exists rather than hard coding an array.